build.clj

```diff
@@ -0,0 +1,59 @@
+(ns build
+  (:refer-clojure :exclude [test])
+  (:require [clojure.tools.build.api :as b]
+            [borkdude.gh-release-artifact :as gh]
+            [org.corfield.build :as bb]))
+
+(def lib 'io.replikativ/konserve)
+(def version (format "0.7.%s" (b/git-count-revs nil)))
+(def class-dir "target/classes")
+(def basis (b/create-basis {:project "deps.edn"}))
+(def jar-file (format "target/%s-%s.jar" (name lib) version))
+
+(defn clean
+  [_]
+  (b/delete {:path "target"}))
+
+(defn jar
+  [opts]
+  (-> opts
+      (assoc :class-dir class-dir
+             :src-pom "./template/pom.xml"
+             :lib lib
+             :version version
+             :basis basis
+             :jar-file jar-file
+             :src-dirs ["src"])
+      bb/jar))
+
+(defn test "Run the tests." [opts]
```

Comment:
Somehow the tests just say `true`. Not sure whether this occurred for you as well.

Reply:
When I run the kaocha tests locally the `true` appears prior to the reflection warnings. It turns out, it actually originates from [this line](https://github.com/replikativ/konserve/blob/d72818fc0c1d25a7dd32fab3d8dd1cf1307b8626/deps.edn#L29).

In fact, with this line, the tests cannot be run via the build script. I suspect that is because a `:main-opt` is already there so the test runner main doesn't get added (see [here](https://github.com/seancorfield/build-clj/blob/main/README.md#running-tests)). 
I find that quite unsatisfying because that means the reflection check has to be taken out of  there and I quite liked that check in that spot. There are still libraries from us that show those warnings and it would be good to detect those things early from now on.